cms-2016-simulated-datasets/code/das_json_store.py

```diff
@@ -0,0 +1,128 @@
+import json
+import os
+import subprocess
+import sys
+import threading
+from time import sleep
+
+from eos_store import check_datasets_in_eos_dir
+from utils import get_dataset_name, get_from_deep_json
+
+
+def get_parent_dataset(dataset, das_dir):
+    "Return parent dataset to the given dataset or an empty string if no parent found."
+    parent_dataset = ''
+
+    filepath = das_dir + '/parent/' + dataset.replace('/', '@') + '.json'
+
+    if os.path.exists(filepath) and os.stat(filepath).st_size != 0:
+        parent_dataset = get_from_deep_json(get_das_store_json(dataset, 'parent', das_dir), 'parent_dataset')
+    return parent_dataset
+
+
+def get_das_store_json(dataset, query='dataset', das_dir=''):
+    "Return DAS JSON from the DAS JSON Store for the given dataset and given query."
+
+    if not dataset:
+        print('[ERROR] There is no DAS JSON store', query, 'for dataset', dataset,
+              file=sys.stderr)
+        return json.loads('{}')
+
+    filepath = das_dir + '/' + query + '/' + dataset.replace('/', '@') + '.json'
+    if os.path.exists(filepath) and os.stat(filepath).st_size != 0:
+        with open(filepath, 'r') as filestream:
+            return json.load(filestream)
+    else:
+        print('[ERROR] There is no DAS JSON store ' + query + ' for dataset ' + dataset,
+              file=sys.stderr)
+        return json.loads('{}')
+
+
+def mydasgoclient(dataset, query, out_dir, out_file):
+    "Interface to dasgoclient"
+   
+    out = out_dir + '/' + query + '/' + out_file
+    if  os.path.exists(out) and os.stat(out).st_size != 0:
+        print('==> {:<9} {}'.format(query, dataset) +
+            '\n==> File already exist, skipping...\n')
+        return
+   
+    print('\t{:<9} {}'.format(query, dataset))
+
+    cmd = 'dasgoclient -query "'
+    if query != "dataset":
+        cmd += query + ' '
+    cmd += 'dataset=' + dataset + '" -json'
```

Comment:
BTW since we are using Python 3, expressions like this can be made more readable by using f-strings:

old:

```python
    cmd += 'dataset=' + dataset + '" -json'
```

new:

```python
    cmd += f'dataset={dataset}" -json'
```

The same technique to format string with variable replacements could be used elsewhere in the code in order to simplify string concatenation etc.